### DISCUSSION TITLE
Fluent UI React-Northstar form

### DISCUSSION THREAD

**Discussion by WynandCoetzerIT** [microsoft/fluentui#15947] [Q&A] [answered]
Hi,

I was hoping that someone could point me to some documentation/demo on how to use a Form in Fluent UI React-Northstar.
I am having some issues with the Datepicker submitting the form when clicked and I cant seem to hook on to the values of any items in the form.

Also, I am currently using a MaskedTextField from office-ui-fabric-react and re-styling it to try and fit the fluent ui look so if there is a way to ad a mask to a Fluent UI input that would be great.

I am creating this app as a Microsoft Teams Tab, so I would like to be able to theme the components in the form.

Any help would be greatly appreciated.

**Comment by levithomason** [comment, marked as answer]
Hello @WynandCoetzerIT, you'll want to have listeners on your form elements and save their value changes in your state. Then, you'll capture the form submission and read the form state at that time.

### 🎉 Solution

Here's a working [codesandbox](https://codesandbox.io/s/sweet-hertz-mkll9) with the setup you described.

>Note, the warning on the Datepicker docs.
>
><img width=500 src="https://user-images.githubusercontent.com/5067638/101667765-95a79480-3a04-11eb-8e8e-a680ed6037f4.png" />
>
>The next release, 0.52.x has may Datepicker features and functionality..

### Other Resources

- Fluent UI React Northstar [Form](https://fluentsite.z22.web.core.windows.net/0.51.3/components/form/definition) docs
- Fluent UI React Northstar [Datepicker](https://fluentsite.z22.web.core.windows.net/0.51.3/components/datepicker/definition) docs
- Office UI Fabric React [MaskedTextField](https://docs.microsoft.com/en-us/javascript/api/office-ui-fabric-react/maskedtextfield?view=office-ui-fabric-react-latest) docs

You also might be interested in React's docs on HTML [forms](https://reactjs.org/docs/forms.html) for idiomatic React patterns.
